lib/genpack.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { encode, decode, ExtensionCodec } = require('@msgpack/msgpack')
+const emojiKeywords = require('emojilib')
+const emojiData = require('unicode-emoji-json')
+
+const keywordsMap = new Map()
+const emmojiInfo = new Map()
+
+const emojiSymbols = Object.keys(emojiKeywords)
+for (const emojiSymbol of emojiSymbols) {
+  emojiKeywords[emojiSymbol].forEach(keyword => {
+    if (keywordsMap.has(keyword) === false) {
+      keywordsMap.set(keyword, [])
+    }
+    keywordsMap.get(keyword).push(emojiSymbol)
+  })
+  emmojiInfo.set(emojiSymbol, {
+    ...emojiData[emojiSymbol],
+    keywords: emojiKeywords[emojiSymbol]
+  })
+}
+
+const extensionCodec = new ExtensionCodec()
+const MAP_EXT_TYPE = 1 // Any in 0-127
+extensionCodec.register({
+  type: MAP_EXT_TYPE,
+  encode: (object) => {
+    if (object instanceof Map) {
+      return encode([...object])
+    } else {
+      return null
+    }
+  },
+  decode: (data) => {
+    const array = decode(data)
+    return new Map(array)
+  }
+})
+const encodedData = encode(
+  {
+    keywords: keywordsMap,
+    emoji: emmojiInfo
+  },
+  { extensionCodec }
+)
+
+const packBuffer = Buffer.from(encodedData.buffer, encodedData.byteOffset, encodedData.byteLength)
+const emojiPack = `
+module.exports = function getPack() {
+  const dataStr = '${packBuffer.toString('base64')}';
+  let dataArray;
+
+  const inJXA = (function() {
+    try {
+      eval('ObjC')
+      return true
+    } catch {
+      return false
+    }
+  }())
+
+  if (inJXA === true) {
+    const encodedData = $.NSData.alloc.initWithBase64EncodedStringOptions(dataStr, 0)
```

Comment:
This would work. As shown in:

```js
ObjC.import('Foundation')

var encodedData = $.NSData.alloc.initWithBase64EncodedStringOptions('Zm9vCg==', 0)
var encodedStr = $.NSString.alloc.initWithDataEncoding(encodedData, $.NSUTF8StringEncoding)
console.log(ObjC.unwrap(encodedStr))

// osascript -l JavaScript index.js
// ^ will output "foo"
```

But something is going wrong with the msgpack decode (out of bounds array access). It may just be easier, and fast enough, to render standard object maps into the `emoji.pack.js` and use those instead of proper `Map` objects.